low-level/files/opt/seagate/sspl/setup/resource_map/storage.py

```diff
@@ -280,6 +281,80 @@ def get_disk_groups_info(self):
                 })
         return dg_data
 
+    def get_sideplane_expanders_info(self):
+        """Update and return sideplane_expanders information."""
+        sideplane_expander_list = []
+        sideplane_expander_data = []
+        enclosures = self.get_realstor_encl_data("enclosures")
+        # TO-DO: Get sideplane_expander data for CORVAULT.
+        encl_drawers = enclosures[0].get("drawers")
+        if encl_drawers:
+            for drawer in encl_drawers:
+                sideplanes = drawer.get("sideplanes")
+                sideplane_expander_list.extend(sideplanes)
+        for sideplane in sideplane_expander_list:
+            sideplane_health = sideplane.get("health", "NA")
+            if sideplane_health in HEALTH_UNDESIRED_VALS:
+                description = "Sideplane is in unknown health state."
+            elif sideplane_health == "OK":
+                description = "Sideplane is in good state."
+            else:
+                description = "Sideplane is not in good state."
+            expanders = sideplane.get("expanders")
+            expander_data = self.get_expander_data(expanders)
+            sideplane_dict = {
+                "uid": sideplane.get("durable-id", "NA"),
+                "fru": "true",
+                "last_updated": int(time.time()),
+                "health": {
+                    "status": sideplane_health,
+                    "description": description,
+                    "recommendation": sideplane.get("health-recommendation", "NA"),
+                    "specifics": [
+                        {
+                            "name": sideplane.get("name", "NA"),
+                            "location": sideplane.get("location", "NA"),
+                            "drawer-id": sideplane.get("drawer-id", "NA"),
+                            "expanders": expander_data
+                        }
+                    ]
+                }
+            }
+            sideplane_expander_data.append(sideplane_dict)
+        return sideplane_expander_data
+
+    def get_expander_data(self, expanders):
```

Comment:
`get_expander_data` can be marked as `staticmethod`

Reply:
Or @jjxsg , @shriya-deshmukh , Just the way we did in server, we can use `get_health_template` and `set_health_data`  functions to make this easy ??

Reply:
@sumedhak27 yes, but to maintain consistency, all FRUs in storage should follow that design. @cdeshmukh is that can be done with this PR itself?

Reply:
Added 'get_heath_template' and 'set_heath_data' function for all FRUs  present in storage file.

Reply:
yes, atlast some one will have to do, which this PR is handling